### PR TITLE
[Teaser] Add support for multiple call to action (CTA) (#155)

### PR DESCRIPTION
**Pull request by vladbailescu**
Initial proposal for multiple CTAs implementation

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/models/Teaser.java [hunk 1]
```diff
@@ -28,6 +31,20 @@
  */
 public interface Teaser extends ComponentExporter {
 
+    /**
+     * Name of the property that will store if the teaser has Call-to-Action elements
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/models/Teaser.java]: I'd write it in present form: 
Name of the property that stores if the teaser has Call-to-Action elements

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/models/Teaser.java [hunk 2]
```diff
@@ -28,6 +31,20 @@
...
+     *
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    String PN_WITH_CTA = "withCTA";
+
+    /**
+     * Name of the node where the Call-to-Action elements will be stored
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/models/Teaser.java]: I'd write it in present form: 
Name of the parent node where the Call-to-Action elements are stored

File: content/src/content/jcr_root/apps/core/wcm/sandbox/components/teaser/v1/teaser/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -22,14 +22,52 @@
                             sling:resourceType="granite/ui/components/coral/foundation/container"
                             margin="{Boolean}true">
                         <items jcr:primaryType="nt:unstructured">
+                            <withCTA
+                                    jcr:primaryType="nt:unstructured"
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/checkbox"
+                                    fieldDescription="When checked, allow adding Call To Action (CTA) elements."
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/sandbox/components/teaser/v1/teaser/_cq_dialog/.content.xml]: I'd reformulate as follows: 
Check to allow adding Call To Action (CTA) elements.

**Comment by vladbailescu** [comment]
Merging this as @gabrielwalt is ok with the looks and behaviour of the dialogs. @andreeadracea will submit a new PR with the final changes in the model and implementation.

**Pull request merged**